ts/utils/hooks/useOnFocus.tsx

```diff
@@ -0,0 +1,42 @@
+import { Millisecond } from "italia-ts-commons/lib/units";
+import { useContext, useEffect, useState } from "react";
+import { NavigationContext } from "react-navigation";
+
+/**
+ * Call the action when the component isFocused() and refreshAfter has passed since the last update.
+ * @param loadAction
+ * @param refreshAfter
+ */
+export const useOnFocus = (
+  loadAction: () => void,
+  refreshAfter: Millisecond | undefined = undefined
+) => {
+  const [lastUpdate, setLastUpdate] = useState<Date | undefined>(undefined);
+
+  const navigation = useContext(NavigationContext);
+  useEffect(() => {
+    const now = new Date();
+    const shouldRefreshDelay =
+      refreshAfter === undefined ||
```

Comment:
As developer if I use `useOnFocus` by passing `refreshAfter===undefined` I expect it runs `loadAction` only once
If this is a good point `useOnFocus` and `useRefreshOnFocus` could be a single function?